### ISSUE TITLE
Implement TwG settings edit UI

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#5531]
#5456 implemented the Thank with Google `SettingsView` component with its actual content. This issue should do the same for the module's `SettingsEdit` component. The content for this editing component largely matches the second step (customization) of the setup flow (see #5529).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The TwG `SettingsEdit` component should receive its actual UI based on [this Figma design](https://www.figma.com/file/t5zDpQ28Lo7ZyRcOpGdqgu/TwG?node-id=1205%3A6424) (or the alternate version with "Position" instead of "Prominence" https://www.figma.com/file/t5zDpQ28Lo7ZyRcOpGdqgu/TwG?node-id=1205%3A7472).
    * For now, only the "Supporter wall widget" section of that UI should be omitted here as it is still being discussed further.
* This UI is almost similar to the one from the second module setup step (customize) which is implemented in #5529, and it all underlies the same requirements and should reuse components implemented as part of that issue. Please review the ACs from that issue as well to have the necessary context.

## Implementation Brief

* In `assets/js/modules/thank-with-google/components/settings/SettingsForm.js`:
    - Add the `<ButtonPlacement>`, `<ColorRadio>` and `<PostTypes>` components created in #5529 to match the Figma mocks. 
    - Ensure the submit button logic as part of the Footer creates the request to successfully store updated settings / option values in the datastore.
    - Ensure `SettingsForm.stories.js` is updated to reflect the new components.

### Test Coverage

* No new tests required.

## QA Brief

* Follow the Setup instructions in the QA Brief for #5450 to connect the TwG module.
* Go to the settings page for the TwG module.
* Change and save TwG settings.
* Remove snippets added from the #5450 ticket
* Verify that saved settings remain the same after you refresh the page.
* Verify that the storybook for the settings for reflects UI changes.

## Changelog entry

* Implement UI for the Thank with Google settings.

**Comment by jimmymadon** [comment]
@tofumatt Apologies I missed that! (I mixed the mock for the setup customize component I was working on before this issue).

@felixarntz Not sure of the ACs for the "Supporter Wall Widget" setting here. #5462 is still in IBR. I guess we might need to define an action to "switch off" the Supporter wall (as the Figma design suggests). This will be in addition to the potential new selector added in #5462 which checks if the wall has been added or not. 

In addition to the above, we are going to wait till next week to finalise the design for the `<PostTypesSelect>` component which ideally should be consistent here and in #5529.

**Comment by tofumatt** [comment]
Thanks for the clarifications and context @felixarntz and @jimmymadon.

In that case, looks good and makes sense!

IB ✅

**Comment by eugene-manuilov** [comment]
There is nothing left to implement here after #5529 is merged. Ready for QA.

**Comment by wpdarren** [comment]
### QA Update: ❌ 

I have a few observations on TwG settings edit UI.  

1.  The Publication ID field appears in settings and I can edit it.  Looking at the figma designs, this field isn't included. 

![image](https://user-images.githubusercontent.com/73545194/182142600-184fbdbd-e719-4c5e-9202-c8c85fc03ab6.png)

2. When I choose the type as `fixed`,  the radio button for 'Manual'  is not clickable, but the entries underneath it, i.e. 
Above the post. Should we display a radio button if we are asking the user to select one of the 3 options below it?  I felt this is a UX/UI issue. 

![image](https://user-images.githubusercontent.com/73545194/182143642-38a0b9ea-1e93-43af-b932-9ddf67ebb899.png)

3. The UI of the Post types looks different when comparing with Figma designs. According to the designs you can remove post types, which doesn't make sense, so figma look wrong.  Thought I would raise and not assume. 😄 

This is what I see on my test site. 
![image](https://user-images.githubusercontent.com/73545194/182144840-2dc00c9f-ad8e-41d4-bb6a-09663788b1b1.png)

4. I suspect that this is expected at this point but thought it was worth checking.  When I edit TwG settings and save, the settings page does not have all of the fields. It just displays PublicationID and Color.  Is this expected? I removed the code from the functions.php which was used to initially connect TwG. 

![image](https://user-images.githubusercontent.com/73545194/182145753-21d89176-199f-48e8-aaf1-5542077525e0.png)

5. On mobile devices i.e. tested this initially on iPhone 12 Pro, the settings screen has styling issues due to the type boxes. 

<details>
 <summary>See mobile screenshot</summary>

![image](https://user-images.githubusercontent.com/73545194/182147471-5586e393-d7f0-40b8-af59-2165146e2d43.png)

</details>

6. There are a few differences with styles between my site and figma designs. I suspect figma is wrong. 
    - The font style on the headings, i.e. type, color etc. is Google Sans Display but on Figma is Google Sans Text. 
    - 

**Verified:**

- The saved settings remain the same after I refresh the page. 

**QA note:** at this point the Supporter wall widget section of TwG is not appearing, as per the AC>

**Comment by jimmymadon** [comment]
Hey @wpdarren, @eugene-manuilov had originally picked this up for execution so I've assigned him for now - but let me know if I should have a look.

**Comment by eugene-manuilov** [comment]
@felixarntz or @aaemnnosttv can anyone of you chime in regarding the publication ID setting on the settings form? The design mockup doesn't show it, but I assume we need this because otherwise users won't be able to change it without re-connecting the module, right?

**Comment by eugene-manuilov** [comment]
> 2\. When I choose the type as `fixed`,  the radio button for 'Manual'  is not clickable, but the entries underneath it, i.e.
>     Above the post. Should we display a radio button if we are asking the user to select one of the 3 options below it?  I felt this is a UX/UI issue.

@wpdarren Matthew created a ticket for it: https://github.com/google/site-kit-wp/issues/5623

**Comment by eugene-manuilov** [comment]
> 3\. The UI of the Post types looks different when comparing with Figma designs. According to the designs you can remove post types, which doesn't make sense, so figma look wrong.  Thought I would raise and not assume.

Yes, I think this deviation is okay because we reused the component that we already have for roles editor in the dashboard sharing settings.

**Comment by eugene-manuilov** [comment]
> 4\. I suspect that this is expected at this point but thought it was worth checking.  When I edit TwG settings and save, the settings page does not have all of the fields. It just displays PublicationID and Color.  Is this expected? I removed the code from the functions.php which was used to initially connect TwG.

Yes, the settings view is a separate ticket which is not completed yet.

**Comment by eugene-manuilov** [comment]
@wpdarren styles issues should be fixed now. Could you please take a look again?

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@eugene-manuilov I have found a few other issues when refreshing the develop branch. 

1. When you select one of the image radio buttons they do not highlight.  Select `Fixed` type and the position fields do appear underneath, but the fixed type isn't highlighted. The `Overlay` type is still highlighted. 

![image](https://user-images.githubusercontent.com/73545194/182377246-0d4dbbba-4ca7-48fe-a18b-2e0dcc3e69b6.png)

2. When you click on `Fixed` type, the `Auto` is no longer selected by default.  See screenshot above. 

3. The `color` styling looks a little messed up as you can see from the screenshot. 

![image](https://user-images.githubusercontent.com/73545194/182377433-60fa411d-381d-454d-8663-da0886060a40.png)

4. The issue mentioned in 1. above, also applies to the `Prominence` image radio buttons.  When I click on `High` the highlighting remains on `Low`

![image](https://user-images.githubusercontent.com/73545194/182378094-6f973543-058e-4fc1-83bc-0d037734ed81.png)

The styling issues mentioned in my previous comment are fixed though!

**Comment by mohitwp** [comment]
@wpdarren 
I was testing #5623 and while comparing I found that in figma design color of the checked radio button is different from the implemented color. I found that this PR may be addressing UI issues so raising here.

![image](https://user-images.githubusercontent.com/94359491/182593542-40eac3a1-eea4-4b52-85bc-bc29793c6c2e.png)

![image](https://user-images.githubusercontent.com/94359491/182593607-a8155485-b112-45c2-b5ae-dcec1c900606.png)

**Comment by wpdarren** [comment]
@mohitwp thanks, I will have a look at that!  Bear in mind we have implemented GM2+ styles so its possible that the figma designs are out of date, I found that a few times with these TwG tickets. Good call though, will look into that.

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@eugene-manuilov  point 2 in my comment above still seems to be an issue:

>  When you click on Fixed type, the Auto is no longer selected by default. 

Previously, the 'Auto' radio button was selected by default but now nothing is defaulted.  Is this expected? 

Also, @mohitwp picked up that the radio button color is not the same as the figma design (which is  #3C7251) is that because all of our other radio buttons are black within the plugin, or, has this been missed?

**Comment by wpdarren** [comment]
> @wpdarren I believe you saw Auto selected by default in the storybook, right? No, there is no default value for it, you need to select manually which placement/prominence you need.

@eugene-manuilov no, this was on a test site when I first started testing this.  If this is the expected behaviour though that's fine. I will get on with approving this ticket.

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@aaemnnosttv just confirming that the point below hasn't been answered. 

> @felixarntz or @aaemnnosttv can anyone of you chime in regarding the publication ID setting on the settings form? The design mockup doesn't show it, but I assume we need this because otherwise users won't be able to change it without re-connecting the module, right? 

This is the only point outstanding and it's ready to approve. 

![image](https://user-images.githubusercontent.com/73545194/182617061-196a085a-44ba-49fb-95f2-599d6e5885a5.png)

**Verified:**

- When settings were changed and saved they remain the same when the page is refreshed. 
- Checked that the changed settings also appear correctly on the settings overview. 
- Checked settings UI to the Figma designs.  Tested on desktop and mobile screen sizes. 
- Verified that the storybook for the settings for reflects UI changes.
- Confirmed that the issues found above were fixed including the styling issues. 

<details>
 <summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/73545194/182617380-9012dfee-f64f-4b8b-8288-bfc395532d9d.png)
![image](https://user-images.githubusercontent.com/73545194/182617485-fbfe1d2e-469b-4537-81db-7f9eef150b6c.png)
![image](https://user-images.githubusercontent.com/73545194/182617542-49886f77-e983-4127-8b1b-8821fd483b14.png)
![image](https://user-images.githubusercontent.com/73545194/182617809-52fdfb90-a5ac-4fc5-8378-72f6af2dde02.png)

</details>

**Comment by aaemnnosttv** [comment]
@wpdarren @eugene-manuilov there should not be an input for the publication ID on the settings yet. The design doc states that [it will not be possible to have more than one TwG publication at first](https://docs.google.com/document/d/17ysrd8ya64T8rAgSriEo7MJxYyLf_nvm7nOn2KR248o/edit#heading=h.s41otwupop9v) so we don't need to worry about the ability to change it at this point.

AdSense is similar in that we don't expose a way to change the connected account ID once set up is completed.

Let's keep the implementation consistent with the design as it is now. If and when we add this capability in the future, it should be informed by a new design.

**Comment by eugene-manuilov** [comment]
@aaemnnosttv do you mind reviewing #5656?

**Comment by aaemnnosttv** [comment]
Thanks @eugene-manuilov – back to you @wpdarren for another pass 👍

**Comment by wpdarren** [comment]
### QA Update:  ✅ 

**Verified:**

- When settings were changed and saved they remain the same when the page is refreshed. 
- Checked that the changed settings also appear correctly on the settings overview. 
- Checked settings UI to the Figma designs.  Tested on desktop and mobile screen sizes. 
- Verified that the storybook for the settings for reflects UI changes.
- Confirmed that the issues found above were fixed including the styling issues. 
- The PublicationID no longer appears when you connect or edit TwG settings. 

**Note:** I did notice an `additional permissions` popup appear when I clicked on the edit settings link.  This did not happen yesterday, and I am unsure if this is to be expected, or, if this or another ticket caused the issue. Have left a emssage with Eugene on Slack for his opinion. In the meantime, I have moved this to approved. 

<details>
 <summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/73545194/182617380-9012dfee-f64f-4b8b-8288-bfc395532d9d.png)
![image](https://user-images.githubusercontent.com/73545194/182617485-fbfe1d2e-469b-4537-81db-7f9eef150b6c.png)
![image](https://user-images.githubusercontent.com/73545194/182617542-49886f77-e983-4127-8b1b-8821fd483b14.png)
![image](https://user-images.githubusercontent.com/73545194/182617809-52fdfb90-a5ac-4fc5-8378-72f6af2dde02.png)
![image](https://user-images.githubusercontent.com/73545194/182618980-4769ab1b-0ca7-4b63-a9bc-c62b7be4faf3.png)
![image](https://user-images.githubusercontent.com/73545194/182619028-d2ff8173-1ddb-43ca-b11e-b50c08950f82.png)
![image](https://user-images.githubusercontent.com/73545194/182618999-61626e2e-372a-4d46-a3c4-42b8c95ce57d.png)
![image](https://user-images.githubusercontent.com/73545194/183067648-e09bc021-175b-450f-a889-f639f0402ffc.png)

</details>

**Comment by felixarntz** [comment]
@wpdarren @eugene-manuilov One small piece that doesn't look optimal here and also in that differs from Figma is that the Prominence preview graphic boxes differ in height, which looks odd on desktop.
<img width="654" alt="Screen Shot 2022-08-11 at 9 07 08 PM" src="https://user-images.githubusercontent.com/3531426/184220081-7c8a4233-6511-4265-80ea-3a7a643772dc.png">

Fixing this right now is not super critical, but if it is straightforward, it would be great to open a follow up PR against `main`. Let me know, otherwise we may need to follow up on it later.

**Comment by wpdarren** [comment]
@felixarntz @eugene-manuilov thanks for highlighting this!  Would be good to fix. I just noticed that the square borders on the colors too, is this a recent change because previously the borders were circle. Something seems amiss with the styling? We had an issue previously with the styling so unsure if its linked. 

![image](https://user-images.githubusercontent.com/73545194/184296783-1a1e728f-f0e6-454e-a91d-5650bc7153c1.png)

**Comment by aaemnnosttv** [comment]
@felixarntz the visual difference in height seems to have been fixed in develop with the update to the images 
<img width="609" alt="image" src="https://user-images.githubusercontent.com/1621608/184693623-b5479254-7dbe-449f-a210-1aec1ed80d49.png">

There is however a very minor shift that happens though when switching between selected choices here. 

I think we can address the remaining touch-up here in https://github.com/google/site-kit-wp/issues/5689 which also includes @wpdarren 's observation above as a Safari-specific detail.
